### DISCUSSION TITLE
Message broker cluster - is it possible?

### DISCUSSION THREAD

**Discussion by hemmokarja** [miguelgrinberg/Flask-SocketIO#2108] [Q&A] [answered]
I'd want to know on a general level, should it be possible to use a cluster of message brokers with Flask-SocketIO, or is there some fundamental reason why this won't work?

I've experimented my fair share with this over the past couple of weeks but with little to no luck. (A simple RabbitMQ cluster with a fanout exchange and one queue per node, and an NLB sitting in front of the cluster.) So before I continue banging my head against the wall any longer, I thought it would be more beneficial to ask here.

I'm not looking for a full guide for how to do this; I'd just want to know whether there's something in the inner workings of Flask-SocketIO or SocketIO that makes the effort futile (or whether it's just my incompetence with RabbitMQ, lol.) ...though, all tips and tricks are much appreciated.

**Comment by miguelgrinberg** [comment, marked as answer]
The architecture of the client managers does not really care about this level of detail, so yes, this is definitely possible. If none of the client manager implementations provided in the python-socketio repository do what you need, you can always create your own subclass. As long as the `_publish()` and `_listen()` methods are implemented for your message queue cluster everything should work.

**Reply by hemmokarja** [reply]
Thanks for the super quick response.

This is great news - I was just about to give up. I guess I just have to try harder :)

Edit: turns out the issue was indeed with my RabbitMQ configuration, and `socketio.KombuManager` can indeed handle a RabbitMQ cluster with a fanout exchange straight out of the box.
